numpy/core/src/multiarray/ctors.c

```diff
@@ -2005,7 +2010,16 @@ PyArray_FromArray(PyArrayObject *arr, PyArray_Descr *newtype, int flags)
             return NULL;
         }
 
-        if (flags & NPY_ARRAY_UPDATEIFCOPY)  {
+        if ((flags & NPY_ARRAY_UPDATEIFCOPY) ||
+            (flags & NPY_ARRAY_UPDATEIFCOPY_CLEAR_B4_EXIT))  {
+            if (flags & NPY_ARRAY_UPDATEIFCOPY) {
+                if (DEPRECATE("Use NPY_ARRAY_UPDATEIFCOPY_CLEAR_B4_EXIT and"
+                        " be sure to call PyArray_ResolveUpdateIfCopy") < 0) {
```

Comment:
Ultimately yeah it should be deprecated, but I thought your plan was to initially introduce this with all the warnings hidden behind a `DEPRECATE_UPDATEIFCOPY` ifdef.